android/src/main/java/com/reactnativechangeicon/ChangeIconModule.kt

```diff
@@ -12,13 +12,30 @@ import com.facebook.react.bridge.ReactMethod
 import com.facebook.react.bridge.Promise
 
 class ChangeIconModule(reactContext: ReactApplicationContext, private val packageName: String) : ReactContextBaseJavaModule(reactContext), Application.ActivityLifecycleCallbacks {
-    private var classesToKill: MutableList<String> = mutableListOf<String>()
-    private var iconChanged: Boolean = false;
+    private var classesToKill: MutableList<String> = mutableListOf()
+    private var iconChanged: Boolean = false
     private var componentClass: String = ""
     override fun getName(): String {
         return "ChangeIcon"
     }
 
+    @ReactMethod
+    fun getIcon(promise: Promise){
+        val activity: Activity? = currentActivity
+        if (activity == null) {
+            promise.reject("activity is null")
+            return
+        }
+        if (componentClass.isEmpty()) componentClass = activity.componentName.className
+
+        if(componentClass.split("MainActivity")[1].isEmpty()){
+            promise.resolve("default")
+        } else {
+            promise.resolve(componentClass.split("MainActivity")[1])
```

Comment:
please store `componentClass.split("MainActivity")[1]` in a variable
then return that variable or "default"